src/rdkafka_broker.c

```diff
@@ -3830,14 +3852,32 @@ static int rd_kafka_broker_fetch_toppars (rd_kafka_broker_t *rkb, rd_ts_t now) {
 
         if (unlikely(rkb->rkb_active_toppar_cnt == 0))
                 return 0;
-
+#if WITH_ZSTD
 	rkbuf = rd_kafka_buf_new_request(
                 rkb, RD_KAFKAP_Fetch, 1,
-                /* ReplicaId+MaxWaitTime+MinBytes+TopicCnt */
-                4+4+4+4+
-                /* N x PartCnt+Partition+FetchOffset+MaxBytes+?TopicNameLen?*/
-                (rkb->rkb_active_toppar_cnt * (4+4+8+4+40)));
-
+                /* ReplicaId+MaxWaitTime+MinBytes+MaxBytes+isolation_level+session_id+session_epoch+TopicCnt */
+                4+4+4+4+4+4+1+4+
+                /* N x PartCnt+Partition+current_leader_epoch+FetchOffset+log_start_offset+MaxBytes+?TopicNameLen?*/
+                (rkb->rkb_active_toppar_cnt * (4+4+4+8+8+4+40))+
+                /* forgotten_topics CNT*/
+                4+
+                /* N x forgotten_topics_data*/
+                (0)
+                );
+#else
+        rkbuf = rd_kafka_buf_new_request(
+            rkb, RD_KAFKAP_Fetch, 1,
+            /* ReplicaId+MaxWaitTime+MinBytes+TopicCnt */
+            4 + 4 + 4 + 4 +
+            /* N x PartCnt+Partition+FetchOffset+MaxBytes+?TopicNameLen?*/
+            (rkb->rkb_active_toppar_cnt * (4 + 4 + 8 + 4 + 40)));
+#endif
+#if WITH_ZSTD
+        if (rkb->rkb_features & RD_KAFKA_FEATURE_ZSTD)
```

Comment:
Remove the existing ApiVersion_set() and replace with
`rd_kafka_buf_ApiVersion_set(rkbuf, ApiVersion, 0);`